### PR TITLE
Isolate and refactor caching logic

### PR DESCRIPTION
**Pull request by douglascamata**
Signed-off-by: Douglas Camata <159076+douglascamata@users.noreply.github.com>

The cache logic has been moved to its own package.

The refactor improves on: 

- Using a typed string for the cache type, to provide some additional safety. 
- Makes it harder to access the underlying string fields of configuration options that are time durations to, to help avoiding mistakes.

## Breaking change 

* This PR changes the SQLite cache type configuration option from `SQLite` to `sqlite`. Helps avoid issues with upper/lower case mistakes.

### CODE REVIEW

File: pkg/cache/config.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	if err := c.load(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// load validates the cache configuration from the parser and copy it
+// into the configuration.
+func (c *Config) load() error {
+	switch c.cacheParser.Type {
```
Comment (bwplotka) [inline pkg/cache/config.go:62]: Can we lower case the Type here, so we can make this change but also without breaking change?

File: pkg/cache/config.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+// NewConfig is the constructor for Config.
+func NewConfig() Config {
+	return Config{
+		cacheParser: newConfigParser(),
+	}
+}
+
+// Present tell whether a cache configuration is present.
+func (c *Config) Present() bool {
```
Comment (bwplotka) [inline pkg/cache/config.go:43]: "present" 


(: 

What about: 
```suggestion
func (c *Config) IsSet() bool {
```

File: pkg/cache/config.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	return c.Type != cacheTypeNone
```
Comment (bwplotka) [inline pkg/cache/config.go:44]: Also 
```suggestion
	return c.Type != cacheTypeNone && c.Type != cacheTypeEmpty
```

**Review by bwplotka — APPROVED** [review]
Thanks! Some nits, otherwise LGTM

File: pkg/cache/config.go (hunk 1 above)
Comment (douglascamata) [reply pkg/cache/config.go:62]: @bwplotka I can un-export the `Type`, but I need a getter to see the actual type. What's the intent of this suggested change?

Comment (douglascamata) [reply pkg/cache/config.go:62]: Ah, nevermind. I was confused.

**Comment by douglascamata** [comment]
@bwplotka feedback taken care of. 👍 🎉

**Review by saswatamcode — APPROVED** [review]
Thanks!

**Pull request merged**